addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,388 @@
+    expense_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.expense_accrual_account_id")
+    revenue_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.revenue_accrual_account_id")
+
+    # change account
+    destination_account_id = fields.Many2one(string="To", comodel_name='account.account', help="Account to transfer to.")
+    display_currency_helper = fields.Boolean(string="Currency Conversion Helper", compute='_compute_display_currency_helper', help="Technical field used to indicate whether or not to display the tooltip informing a currency conversion will be performed with the transfer")
+
+    @api.constrains('percentage', 'action')
+    def _constraint_percentage(self):
+        for record in self:
+            if not (0.0 < record.percentage <= 100.0):
+                raise UserError(_("Percentage must be between 0 and 100"))
+            if record.percentage != 100 and record.action != 'change_period':
+                raise UserError(_("Percentage can only be set for Change Period method"))
+
+    @api.depends('percentage', 'move_line_ids')
+    def _compute_total_amount(self):
+        for record in self:
+            percentage = record.percentage or 100
+            total_amount = percentage * sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) / 100
+            if record.total_amount != total_amount:  # Prevent looping with _compute_percentage
+                record.total_amount = total_amount   # We don't touch/set record.total_amount if not needed
+            record.account_type = 'income' if sum(record.move_line_ids.mapped('balance')) < 0 else 'expense'
+
+    @api.depends('total_amount', 'move_line_ids')
+    def _compute_percentage(self):
+        for record in self:
+            record.percentage = record.total_amount / sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) * 100
+            record.percentage = record.percentage or 100
+
+    @api.depends('destination_account_id')
+    def _compute_display_currency_helper(self):
+        for record in self:
+            record.display_currency_helper = bool(record.destination_account_id.currency_id)
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if self.env.context.get('active_model') != 'account.move.line' or not self.env.context.get('active_ids'):
+            raise UserError(_('This can only be used on journal items'))
+        move_line_ids = self.env['account.move.line'].browse(self.env.context['active_ids'])
+        res['move_line_ids'] = move_line_ids.ids
```

Comment:
don't you need an ORM `[(6, 0, ...)]` command instead?

Reply:
Good question. Both work, but which one is better practice? I guess the explicit ORM command is better...

Reply:
Idk. Keep the one you like xD